game-core/src/main/java/games/strategy/engine/lobby/client/login/ChangePasswordPanel.java

```diff
@@ -0,0 +1,161 @@
+  private final JButton okButton = new JButton("OK");
+
+  private ChangePasswordPanel() {
+    layoutComponents();
+    setupListeners();
+  }
+
+  /**
+   * Creates a new instance of the {@code CreateUpdateAccountPanel} class that is used to update the
+   * specified lobby account.
+   *
+   * @return A new {@code CreateUpdateAccountPanel}.
+   */
+  public static ChangePasswordPanel newChangePasswordPanel() {
+    return new ChangePasswordPanel();
+  }
+
+  private void layoutComponents() {
+    setLayout(new BorderLayout());
+    final JLabel label = new JLabel(new ImageIcon(Util.getBanner(title)));
+    add(label, BorderLayout.NORTH);
+
+    final JPanel main = new JPanel();
+    add(main, BorderLayout.CENTER);
+    main.setBorder(BorderFactory.createEmptyBorder(10, 10, 10, 10));
+    main.setLayout(new GridBagLayout());
+    main.add(
+        new JLabel("Password:"),
+        new GridBagConstraints(
+            0,
+            0,
+            1,
+            1,
+            0.0,
+            0.0,
+            GridBagConstraints.WEST,
+            GridBagConstraints.NONE,
+            new Insets(5, 0, 0, 0),
+            0,
+            0));
+    main.add(
+        passwordField,
+        new GridBagConstraints(
+            1,
+            0,
+            1,
+            1,
+            1.0,
+            0.0,
+            GridBagConstraints.WEST,
+            GridBagConstraints.HORIZONTAL,
+            new Insets(5, 5, 0, 0),
+            0,
+            0));
+    main.add(
+        new JLabel("Confirm Password:"),
+        new GridBagConstraints(
+            0,
+            1,
+            1,
+            1,
+            0.0,
+            0.0,
+            GridBagConstraints.WEST,
+            GridBagConstraints.NONE,
+            new Insets(5, 0, 0, 0),
+            0,
+            0));
+    main.add(
+        passwordConfirmField,
+        new GridBagConstraints(
+            1,
+            1,
+            1,
+            1,
+            1.0,
+            0.0,
+            GridBagConstraints.WEST,
+            GridBagConstraints.HORIZONTAL,
+            new Insets(5, 5, 0, 0),
+            0,
+            0));
+
+    final JPanel buttons = new JPanel();
+    add(buttons, BorderLayout.SOUTH);
+    buttons.setBorder(BorderFactory.createEmptyBorder(10, 5, 10, 5));
+    buttons.setLayout(new FlowLayout(FlowLayout.RIGHT, 5, 0));
+    buttons.add(okButton);
+    okButton.setEnabled(false);
+  }
+
+  private void setupListeners() {
+    okButton.addActionListener(e -> close());
+
+    SwingComponents.addEnterKeyListener(this, this::close);
+
+    DocumentListenerBuilder.attachDocumentListener(
+        passwordField, () -> okButton.setEnabled(validatePasswords()));
+    DocumentListenerBuilder.attachDocumentListener(
+        passwordConfirmField, () -> okButton.setEnabled(validatePasswords()));
+  }
+
+  private void close() {
+    if (validatePasswords() && dialog != null) {
+      dialog.dispose();
```

Comment:
Hmm this seems like a smell to me.
What about registering listeners when the dialog is shown, and unregistering them afterwards?

Reply:
We only really need to make the dialog not visible. The dialog creation code is notable, after we set the dialog to visible it is blocking until no longer visible. The next line then disposes the dialog. 

I updated this method to be a more simple 'not visible': ed4157d2c
Checking out the dispose code there is quite a lot there, probably better then to not 'double dispose'.

Overall it is a pattern for swing JDialog to work like this, essentially to have a sequence like:
```
dialog.setVisible(true);
dialog.dispose();
```